src/main/resources/META-INF/rewrite/apache-httpclient-4-5.yml

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright 2023 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+########################################################################################################################
+# Apache HttpClient 4.5.x
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.apache.httpclient4.UpgradeApacheHttpClient_4_5
+displayName: Migrates to ApacheHttpClient 4.5.x
+description: >
+  Migrate applications to the latest Apache HttpClient 4.5.x release. This recipe will modify an
+  application's build files, make changes to deprecated/preferred APIs, and migrate configuration settings that have
+  changes between versions.
+tags:
+  - apache
+  - httpclient
+recipeList:
+  - org.openrewrite.java.dependencies.UpgradeDependencyVersion:
+      groupId: org.apache.httpcomponents
+      artifactId: httpclient
+      newVersion: 4.5.x
+      overrideManagedVersion: true
```

Comment:
Why the override here? I imagine that might accidentally pin a managed version that would otherwise be fixed with a parent upgrade. Is there anything in the 4.5.x range that requires an upgrade to 4.5.x specifally, like replacement classes added in the last versions?

Reply:
If override of managed versions is not strictly necessary I'd opt for the default here.
```suggestion
```

Reply:
The idea is to migrate to the latest version of 4.5.x before moving to 5.x, since there are some extra deprecated classes and their replacements in 4.5.x that disappear in 5.x

Reply:
We are also overriding managed versions when moving to 5.x. I guess it's either necessary on both places, or neither. WDYT?

Reply:
I try not to pin versions when possible, as that might get projects in a state that's slightly more cumbersome to move away from, if they use a shared or Spring Boot parent for managed versions. And since we change the GAV in the upgrade from 4 to 5 I don't think we need to override managed versions there either.